pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -346,6 +347,29 @@ public StringResultResponse getTaskStateDeprecated(
     }
   }
 
+  @GET
+  @Path("/tasks/all")
+  @Authenticate(AccessType.READ)
+  @ApiOperation("Read all tasks status")
+  public Map<String, TaskState> getAllTasks() {
+    Map<String, TaskState> taskStates = new HashMap<>();
+    _pinotHelixTaskResourceManager.getTaskTypes()
+        .forEach(taskType -> taskStates.putAll(_pinotHelixTaskResourceManager.getTaskStates(taskType)));
+    return taskStates;
+  }
+
+  @POST
+  @Path("/tasks/create")
+  @Authenticate(AccessType.CREATE)
```

Comment:
Instead of taking a generic map, can we add specific arguments here? The arguments can be self-documented instead of the user learning one argument at a time based on the exception thrown.

Or, maybe this can be a table API (since we always want to run tasks on a particular table)

Reply:
Will change this method to mandate taskType, table, taskName with the extra map of configs.
```
public Map<String, String> executeAdhocTask(
      @ApiParam(value = "Task type", required = true) @PathParam("taskType") String taskType,
      @ApiParam(value = "Table name", required = true) @PathParam("tableName") String tableName,
      @ApiParam(value = "taskName name is unique within a task type", required = true) @PathParam("taskName")
          String taskName, Map<String, String> taskConfigs) {
```

Reply:
maybe make `taskName` an optional @QueryParam. And auto-gen a task name if not provided by user.

Reply:
I'd suggest adding a JSON serializable wrapper class to wrap all the arguments and the config map. See `Instance` as an example. The API can be as simple as `POST /tasks/execute`

Reply:
Updated with a AdhocTaskConfig